semtypes/src/main/java/io/ballerina/types/definition/FunctionDefinition.java

```diff
@@ -59,6 +59,15 @@ private SemType createSemType(Atom rec) {
 
     public SemType define(Env env, SemType args, SemType ret) {
         FunctionAtomicType atomicType = FunctionAtomicType.from(args, ret);
+        return defineInner(env, atomicType);
+    }
+
+    public SemType defineGeneric(Env env, SemType args, SemType ret) {
+        FunctionAtomicType atomicType = FunctionAtomicType.genericFrom(args, ret);
+        return defineInner(env, atomicType);
+    }
+
+    private SemType defineInner(Env env, FunctionAtomicType atomicType) {
```

Comment:
If I am not mistaken we are using `xxxInner()` methods to indicate the return semtype is a `cellInner()` value. i.e. return semtype is a `ty` inside a cell. To avoid any possible confusion, how about calling this `defineInternal()`?
```suggestion
    private SemType defineInternal(Env env, FunctionAtomicType atomicType) {
```